source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/overview/products/item/tasks.tsx

```diff
@@ -13,70 +13,137 @@ import { useCurrentUser } from '~/data/containers/with-current-user';
-    // so a subsequent task will become a duplicate because both received tasks
-    // have yet to be added to the local cache.
-    tasks = uniqBy(tasks, (task) => (task.keys || {}).remoteId);
-  } catch (e) {
-    console.debug(
-      'error occurred',
-      e,
-      'we probably need to PR to orbit.js, as this is a race condition scenario that causes the error',
-      'variables that are in use that may be related to this error: ',
-      ['product: ', product, 'currentUser: ', currentUser]
-    );
-  }
+  useEffect(() => {
+    async function fetcher() {
+      let response = await authenticatedGet(`/api/products/${productRemoteId}/transitions/active`);
+      try {
+        let json = await handleResponse(response, t);
+        let data = json.data;
+        setTransition(data);
+        setFetchComplete(true);
+      } catch (e) {
+        setFetchComplete(true);
+      }
+    }
+    fetcher();
+  }, [productRemoteId]);
 
   const { relativeTimeAgo } = useTimezoneFormatters();
   const { pathToWorkflow } = useUserTaskHelpers();
+  const getCurrentUserTask = () => {
+    let tasks = [];
+    try {
+      tasks = dataStore.cache.query((q) =>
+        q.findRecords('userTask').filter({ relation: 'product', record: product })
+      );
+      // for some reason the backend is sending task creations to the frontend qucik
+      // enough where the key-checking mechanisms hasn't yet added the first task,
+      // so a subsequent task will become a duplicate because both received tasks
+      // have yet to be added to the local cache.
+      tasks = uniqBy(tasks, (task) => (task.keys || {}).remoteId);
+      let foundCurrentUser = false;
+      let workTask = null;
+      {
+        tasks.map((task) => {
+          const user = dataStore.cache.query((q) => q.findRelatedRecord(task, 'user'));
+          const taskUserId = remoteIdentityFrom(dataStore, user).keys.remoteId;
+          const currentUserId = remoteIdentityFrom(dataStore, currentUser).keys.remoteId;
 
-  if (tasks.length === 0) {
-    return <div className='w-100 p-sm p-b-md m-l-md fs-13'>{t('tasks.noTasksTitle')}</div>;
+          if (taskUserId === currentUserId) {
+            foundCurrentUser = true;
+            workTask = task;
+          } else if (!workTask) {
+            // If there isn't a return task set, set it to this entry so that an elapsed time
+            // can be calculated, even if there isn't one for the current user.
+            workTask = task;
+          }
+        });
+      }
+      return [foundCurrentUser, workTask];
+    } catch (e) {
+      console.debug(
+        'error occurred',
+        e,
+        'we probably need to PR to orbit.js, as this is a race condition scenario that causes the error',
+        'variables that are in use that may be related to this error: ',
+        ['product: ', product, 'currentUser: ', currentUser]
+      );
+    }
+  };
+  const formatTaskLine = (transition) => {
```

Comment:
this should probably be a utility function as it doesn't seem to have any bearing on the component

Reply:
I moved it back inline and redid the return to format the line in a more conventional way.